pkg/controller/gittrack/gittrack_controller_test.go

```diff
@@ -324,62 +384,74 @@ var _ = Describe("GitTrack Suite", func() {
-				},
-				Spec: farosv1alpha1.GitTrackSpec{
-					SubPath:    path,
-					Repository: repositoryURL,
-					DeployKey:  farosv1alpha1.GitTrackDeployKey{},
-					Reference:  "51798af1c1374d1d375a0eb7a3e53dd67ac5d135",
-				},
-			}
+	const timeout = time.Second * 5
+	const filePathRegexp = "^[a-zA-Z0-9/\\-\\.]*\\.(?:yaml|yml|json)$"
+	const doesNotExistPath = "does-not-exist"
+	const repeatedReference = "448b39a21d285fcb5aa4b718b27a3e13ffc649b3"
 
-			Expect(c.Create(context.TODO(), gt)).NotTo(HaveOccurred())
-			req := reconcile.Request{
-				NamespacedName: types.NamespacedName{
-					Name:      "test",
-					Namespace: "default",
-				},
-			}
-			Eventually(requests, timeout).Should(Receive(Equal(req)))
+	var createInstance = func(gt farosv1alpha1.GitTrackInterface, ref string) {
+		spec := gt.GetSpec()
+		spec.Reference = ref
+		gt.SetSpec(spec)
+		err := c.Create(context.TODO(), gt)
+		Expect(err).NotTo(HaveOccurred())
+	}
 
-			files, err = reconciler.getFiles(gt)
-			Expect(err).ToNot(HaveOccurred())
+	BeforeEach(func() {
+		var err error
+		cfg.RateLimiter = flowcontrol.NewFakeAlwaysRateLimiter()
+		mgr, err = manager.New(cfg, manager.Options{
+			Namespace:          farosflags.Namespace,
+			MetricsBindAddress: "0", // Disable serving metrics while testing
 		})
+		Expect(err).NotTo(HaveOccurred())
+		c = mgr.GetClient()
 
-		AfterEach(func() {
-			Expect(c.Delete(context.TODO(), gt)).NotTo(HaveOccurred())
-		})
+		var recFn reconcile.Reconciler
+		var opts *reconcileGitTrackOpts
+		r, opts = newReconciler(mgr)
+		recFn, requests = SetupTestReconcile(r)
+		opts.clusterGitTrackMode = farosflags.CGTMDisabled
+		Expect(add(mgr, recFn, opts)).NotTo(HaveOccurred())
+		stop = StartTestManager(mgr)
+		instance = &farosv1alpha1.ClusterGitTrack{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "example",
+				Namespace: "",
+			},
+			Spec: farosv1alpha1.GitTrackSpec{
+				Repository: repositoryURL,
+			},
+		}
+	})
 
-		It("Filters files by SubPath", func() {
-			for filePath := range files {
-				Expect(filePath).To(HavePrefix(strings.TrimPrefix(path, "/")))
-			}
-		})
+	AfterEach(func() {
+		close(stop)
+		testutils.DeleteAll(cfg, timeout,
+			&farosv1alpha1.GitTrackList{},
+			&farosv1alpha1.ClusterGitTrackList{},
+			&farosv1alpha1.GitTrackObjectList{},
+			&farosv1alpha1.ClusterGitTrackObjectList{},
+			&v1.EventList{},
+		)
+	})
 
-		It("Filters files by file extension", func() {
-			for filePath := range files {
-				Expect(filePath).To(MatchRegexp(filePathRegexp))
-			}
+	Context("When a ClusterGitTrack resource is created", func() {
+		BeforeEach(func() {
+			createInstance(instance, "a14443638218c782b84cae56a14f1090ee9e5c9c")
 		})
-
-		It("Fetches all files recursively from the SubPath", func() {
-			Expect(files).To(HaveLen(count))
+		It("should not reconcile it", func() {
+			// use a custom timeout here, so that the tests doesn't run too slow on
+			// success
+			Consistently(requests, 1*time.Second).ShouldNot(Receive())
```

Comment:
I'd make this timeout a const to match other test files
```suggestion
			Consistently(requests, consistentlyTimeout).ShouldNot(Receive())
```